website/src/pages/v1/serve/deployment/docker.mdx

```diff
@@ -4,4 +4,93 @@ Docker is a tool that allows you to package an application and its dependencies
 that can run on any system. This makes it easy to deploy applications in a consistent and
 reproducible way, regardless of the underlying infrastructure.
 
-`TODO`
+## Running with Docker
+
+To simplify running GraphQL Mesh, you can use the Docker image and the Docker Compose template we
+provide. This setup allows you to easily configure and run the gateway without the need to install
+Node.js and the required mesh npm packages.
+
+### Prerequisites
+
+Ensure you have Docker and Docker Compose installed on your machine. If not, you can download and
+install them from the official Docker website.
+
+### Docker Compose Setup
+
+1. Create a `docker-compose.yml` file with the following:
+
+```yaml
+version: '3.8'
+
+services:
+  graphql-mesh:
+    image: ghcr.io/ardatan/graphql-mesh:v1
+    ports:
+      - '4000:4000'
+    # Uncomment the line below to add environment variables (ex: for using Graphql Hive integration)
+    # environment:
+    #   HIVE_CDN_ENDPOINT: SECRET
+    #   HIVE_CDN_KEY: SECRET
+    #   HIVE_REGISTRY_TOKEN: SECRET
+    security_opt:
+      - no-new-privileges:true
+    cap_drop:
+      - ALL
+    cap_add:
+      - CHOWN
+    read_only: true
+    volumes:
+      - /tmp # make /tmp writable
+      # Uncomment the line below to inject your custom mesh.config.ts
+      # - ./path/to/local/mesh.config.ts:/app/mesh.config.ts:ro
+```
+
+2. Run Docker Compose:
+
+In the directory containing your `docker-compose.yml` file, run the following command:
+
+```sh
+docker compose up
+```
+
+This command will:
+
+- Pull the GraphQL Mesh Gateway image if it is not already available locally.
+- Pull the supergraph from your Hive registry using the provided environment variables.
+- Start the Mesh Gateway service on port `4000`.
+
+## Custom Docker Image
+
+You can create a custom Docker image that inherits from the `ghcr.io/ardatan/graphql-mesh:v1` base
+image.
+
+```dockerfile
+FROM ghcr.io/ardatan/graphql-mesh:v1
+
+# Extend here
+```
+
+Then build your image:
+
+```sh
+docker build -t graphql-mesh-gateway .
+```
+
+### Running the Docker Container
+
+To run the Docker container with your custom image, use the following command:
+
+```sh
+docker run --cap-drop=ALL --cap-add=CHOWN --security-opt no-new-privileges -p 4000:4000 graphql-mesh-gateway
+```
```

Comment:
this should be before the docker compose instructions.

Reply:
What do these do?
```
--cap-drop=ALL --cap-add=CHOWN --security-opt no-new-privileges
```

Reply:
these are security options, right below it is the description for each option and what they do

Reply:
IMHO we shouldnt overcomplicate or recommend any security trick here. A simple `docker run -p 4000:4000 graphql-mesh-gateway` should suffice. Mainly because security practices change and often depend on user's requirements.

Reply:
I agree with @enisdenjo 
The security stuff should be baked into the image, and then developer can add stuff on top.